pkgs/dart_model/lib/src/dart_model.dart

```diff
@@ -3,12 +3,63 @@
 // BSD-style license that can be found in the LICENSE file.
 
 import 'dart_model.g.dart';
-import 'json_buffer/json_buffer_builder.dart';
+import 'json_buffer/json_buffer_builder.dart' hide Type;
 import 'lazy_merged_map.dart';
 import 'scopes.dart';
 
 export 'dart_model.g.dart';
 
+extension Decl on Declaration {
+  /*WireType get wireType {
+    final path = MacroScope.current.model.pathOf(node)!;
+    final secondLast = path[path.length - 2];
+    switch (secondLast) {
+      case 'members':
+        return WireType.member;
+      case 'scopes':
+        return WireType.interface;
+      default:
+        throw UnsupportedError('Unsupported path: $path');
+    }
+  }*/
+
+  WireType get wireType => Cast.fromJson(node).wireType;
+
+  Cast get cast => Cast.fromJson(node);
+}
+
+enum WireType {
+  member,
+  interface;
+}
+
+extension type Cast.fromJson(Map<String, Object?> node) {
+  WireType get wireType {
+    final path = MacroScope.current.model.pathOf(node)!;
+    final secondLast = path[path.length - 2];
+    switch (secondLast) {
+      case 'members':
+        return WireType.member;
+      case 'scopes':
+        return WireType.interface;
+      default:
+        throw UnsupportedError('Unsupported path: $path');
+    }
+  }
+
+  bool get isMember => wireType == WireType.member;
+  Member? get maybeMember => isMember ? asMember : null;
```

Comment:
nit: maybe `ifMember`? I don't feel strongly either way though.

We probably also want to avoid looking up `wireType` twice here.